interview-questions.md

```diff
@@ -15,54 +15,83 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 
   Researched answer:
+   
+    An index is basically a chronological ranking system for the characters in a string. So if I was using command terminal and input: 
+    console.log("Not another assessment!" [7]), we'd get an output of "h" because when it concerns a string the letters or characters start at 0 and not the traditional 1; thus by counting from 0 to 7, the computer would land on the letter "h" or the 7th character using the provided index. A value can be descripted a number or a set of characters in a string that typically have a comparative relationship. In other words, if I was scripting code to layout parameters to ensure certain inputs would get specific outputs, I would write in key values and the relationship they have with one another to ensure the desired outputs are being produced. 
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: 
+
+  It's a situation that is repeating itself time and time again. So for instance, if a developer was using command terminal and continuously getting the same output from the sequences they put in that, it could be said that's an iteration. 
 
   Researched answer:
 
+  In the realm of Javascript, loops are used to allow programs to perform repetitive tasks. When performing such tasks the programm will go through cycles of this repeating process seen has designated loop cycles, more commonly referred to as loop iterations. Loop iterations are initiated by an iterator, typically the "for" in our line of code (as shown in the example below): 
 
+  for (let i = 0; i < 77; i++) {
+    console.log (i)
+  }
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
 
-  Researched answer:
+  Proper pair programming consist of a driver and navigator. The driver is the one who is writing or executing the code; while the navigator is the one leading what is to be written/typed. Each one role should be respected and both parties should be paying attention to the other so that mistakes are found or avoid as much as possible. It also important to be respectful to the other individual(s) that one is working with as we can all learn from one another and have something to offer the process.
 
+  Researched answer:
 
+  Pair programming is practice of collaborating with one other person on programming tasks in conjoint-fashion. The process consist of a driver and a navigator. The driver is typing out the various lines of code, while the navigator is the one expressing or verbally guiding the driver on what to input next. Both parties however can and should contribute ideas for when troubleshooting occurs and maintain a sense of respect when sharing ideas, corrections, and general communication. These roles should be occassionally swapped so that all parties have experience with both roles. 
```

Comment:
Excellent Steven!  You mentioned you had done some writing in the past and it shows!  Great job of interpreting your research and making your answers very conversational.  Bonus for using examples!